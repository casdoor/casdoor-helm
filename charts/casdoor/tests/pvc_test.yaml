suite: test pvc
templates:
- pvc.yaml

tests:
- it: should not render when persistence is disabled
  set:
    persistence.enabled: false
  asserts:
  - hasDocuments:
      count: 0

- it: should render with default values when persistence is enabled
  set:
    persistence.enabled: true
  asserts:
  - matchSnapshot: {}

- it: should set custom storage class and size
  set:
    persistence.enabled: true
    persistence.storageClass: "fast"
    persistence.size: "10Gi"
  asserts:
  - matchSnapshot: {}
