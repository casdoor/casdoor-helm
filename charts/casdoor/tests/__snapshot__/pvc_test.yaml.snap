should render with default values when persistence is enabled:
  1: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor-data
    spec:
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 8Gi
should set custom storage class and size:
  1: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor-data
    spec:
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 10Gi
      storageClassName: fast
